src/char/int_rodex.c

```diff
@@ -0,0 +1,629 @@
+	int8 opentype = RFIFOB(fd, 11);
+	int64 mail_id = RFIFOQ(fd, 12);
+	struct rodex_maillist mails = { 0 };
+
+	VECTOR_INIT(mails);
+	if (flag == 0)
+		count = inter_rodex_fromsql(char_id, account_id, opentype, &mails);
+	else
+		count = inter_rodex_refresh_fromsql(char_id, account_id, opentype, mail_id, &mails);
+	mapif->rodex_sendinbox(fd, char_id, opentype, flag, count, &mails);
+}
+
+/*==========================================
+* Checks if there are new mails
+*------------------------------------------*/
+void mapif_rodex_sendhasnew(int fd, int char_id, bool has_new)
+{
+	WFIFOHEAD(fd, 7);
+	WFIFOW(fd, 0) = 0x3897;
+	WFIFOL(fd, 2) = char_id;
+	WFIFOB(fd, 6) = has_new;
+	WFIFOSET(fd, 7);
+}
+
+void mapif_parse_rodex_checkhasnew(int fd)
+{
+	int char_id = RFIFOL(fd, 2);
+	int account_id = RFIFOL(fd, 6);
+	bool has_new = inter_rodex->hasnew(char_id, account_id);
+
+	mapif->rodex_sendhasnew(fd, char_id, has_new);
+}
+
+/*==========================================
+ * Update/Delete mail
+ *------------------------------------------*/
+void mapif_parse_rodex_updatemail(int fd)
+{
+	int64 mail_id = RFIFOL(fd,2);
+	int8 flag = RFIFOB(fd, 10);
+
+	switch (flag) {
+	case 0: // Read
+		if (SQL_ERROR == SQL->Query(inter->sql_handle, "UPDATE `%s` SET `is_readed` = 1 WHERE `id` = '%"PRId64"'", rodex_db, mail_id))
+			Sql_ShowDebug(inter->sql_handle);
+		break;
+	
+	case 1: // Get Zeny
+		if (SQL_ERROR == SQL->Query(inter->sql_handle, "UPDATE `%s` SET `zeny` = 0, `type` = `type` & (~2) WHERE `id` = '%"PRId64"'", rodex_db, mail_id))
+			Sql_ShowDebug(inter->sql_handle);
+		break;
+	
+	case 2: // Get Items
+		if (SQL_ERROR == SQL->Query(inter->sql_handle, "DELETE FROM `%s` WHERE `mail_id` = '%"PRId64"'", rodex_item_db, mail_id))
+			Sql_ShowDebug(inter->sql_handle);
+		if (SQL_ERROR == SQL->Query(inter->sql_handle, "UPDATE `%s` SET `zeny` = 0, `type` = `type` & (~4) WHERE `id` = '%"PRId64"'", rodex_db, mail_id))
+			Sql_ShowDebug(inter->sql_handle);
+		break;
+
+	case 3: // Delete Mail
+		if (SQL_ERROR == SQL->Query(inter->sql_handle, "DELETE FROM `%s` WHERE `id` = '%"PRId64"'", rodex_db, mail_id))
+			Sql_ShowDebug(inter->sql_handle);
+		if (SQL_ERROR == SQL->Query(inter->sql_handle, "DELETE FROM `%s` WHERE `mail_id` = '%"PRId64"'", rodex_item_db, mail_id))
+			Sql_ShowDebug(inter->sql_handle);
+		break;
+	}
+}
+
+/*==========================================
+ * Send Mail
+ *------------------------------------------*/
+void mapif_rodex_send(int fd, int sender_id, int receiver_id, int receiver_accountid, bool result)
+{
+	WFIFOHEAD(fd,15);
+	WFIFOW(fd,0) = 0x3898;
+	WFIFOL(fd,2) = sender_id;
+	WFIFOL(fd,6) = receiver_id;
+	WFIFOL(fd,10) = receiver_accountid;
+	WFIFOB(fd,14) = result;
+	WFIFOSET(fd,15);
+}
+
+void mapif_parse_rodex_send(int fd)
+{
+	struct rodex_message msg;
+
+	if(RFIFOW(fd,2) != 4 + sizeof(struct rodex_message))
+		return;
+
+	memcpy(&msg, RFIFOP(fd,4), sizeof(struct rodex_message));
+	if(msg.receiver_id > 0 || msg.receiver_accountid > 0)
```

Comment:
in normal use it shouldn't be, in case it happens, msg.id will stay in 0 and `false` will be sent as result to whoever made this request